Gems/Terrain/Assets/Shaders/Terrain/TerrainCommon.azsli

```diff
@@ -95,25 +95,19 @@ float2 GetWorldXYPosition(in ObjectSrg::PatchData patchData, in float2 vertexPos
     return float2(patchData.m_xyTranslation + vertexPosition * patchData.m_xyScale);
 }
 
-bool IsVertexOutsideOfTerrainBounds(float2 position, float2 minBounds, float2 maxBounds)
-{
-    return (any(position < minBounds) || any(position > maxBounds));
-}
-
 void CalculateTerrainPosition(
     in SceneSrg::TerrainWorldData worldData, in ObjectSrg::PatchData patchData, in VertexInput input,
     out float3 worldPosition, out float4 position, out float2 cullDistance, out float3 normal)
 {
     float2 worldXyPosition = GetWorldXYPosition(ObjectSrg::m_patchData, input.m_xyPosition);
-    if (IsVertexOutsideOfTerrainBounds(worldXyPosition, worldData.m_min.xy, worldData.m_max.xy))
+    if (input.m_zPosition == 1.0)
     {
-        // Output a NaN to remove this vertex.
+        // 1.0 means there is no height data at this vertex, so output a NaN to remove this vertex.
         position = 1.0 / 0.0;
```

Comment:
Since you mention this has been around for a very long time I'll go ahead and approve the PR, since this was the only sticky issue, the rest is just nit stuff.

But I'd also like to know what the Atom guys' feedback is on this.

Reply:
Thanks @invertednormal, nice trick to learn then :)